Support in-page-search in trace view on key=value pairs

bobrik opened the pull request:
Previously one could only filter on `key` or `value`. This wasn't useful when one needed to find spans with `result=false` in a trace view.

With this change one can use `key=value` as a search term, for example:

* `http.method=GET`

Before (no matches):

<img width="1231" alt="image" src="https://user-images.githubusercontent.com/89186/236012512-f97a67e8-8bf8-4372-b698-d0f34383b441.png">

After (a successful match):

<img width="1231" alt="image" src="https://user-images.githubusercontent.com/89186/236012540-7c011a81-ca63-41d2-9051-cad4c5eab02e.png">

yurishkuro commented:
a bit dirty (i.e. would also match on `value="a=b"`), but seems useful regardless. +1

yurishkuro commented:
Overall the in-page search functionality is not very discoverable. When it was just a direct string match it was less of an issue, but if supporting a=b format it would be good to have some help tip somewhere.

Q: in the backend search form we support multiple space-separated tag queries, would be nice to allow the same syntax for in-page search. I believe we even have a parser for the syntax in the UI.

bobrik commented:
I added some tests for the `key=value` syntax.

Here's the tooltip I added in the second commit:

<img width="665" alt="image" src="https://user-images.githubusercontent.com/89186/236377280-c87535d9-a77c-4de5-8039-2fba62699ba7.png">

Using the same syntax as the tag filter on the search page would mean that we lose `-field` syntax, so it might not be a desired change. Trace view search also looks at more things than just tags, which might not be compatible with what we have in the regular search. I'd also say that it's outside of my expertise, since I'm not a frontend person.

bobrik commented:
@yurishkuro, any comments on the latest update?

yurishkuro commented on packages/jaeger-ui/src/components/TracePage/TracePageHeader/TracePageSearchBar.tsx:
```diff
@@ -56,49 +56,79 @@ export function TracePageSearchBarFn(props: TracePageSearchBarProps & { forwarde
+          To preclude certain key-value pairs from participating in the matching, prefix the key
+          with the minus <code>'-'</code> sign, e.g., <code>-http.status_code</code>.
+        </p>
+      </div>
```
I rewrote the tooltip to make the explanation a bit more clear and to remove the styling applied to some key words which was difficult to read on the dark background.

yurishkuro commented:
Sorry it took a while to review, a busy month.
